src/util/Injected.js

```diff
@@ -146,7 +150,7 @@ exports.LoadUtils = () => {
         const newMsgId = new window.Store.MsgKey({
             fromMe: true,
             remote: chat.id,
-            id: window.Store.genId(),
+            id: options.messageId || window.WWebJS.getNewId(),
```

Comment:
Hi, sorry for delay. At this file I use the message Id into options to send a message with a previous generated id. In Client.js I create the function `getNewId` to generate before send the message.

![Captura de Tela 2021-09-01 às 09 44 58](https://user-images.githubusercontent.com/231887/131673415-5340183c-a1a4-4f06-b5ff-9f79136c7313.png)

And in `sendMessage` function, pass the messageId option to process at Injector file, as you as mentioned.
![Captura de Tela 2021-09-01 às 09 50 16](https://user-images.githubusercontent.com/231887/131674150-39d780e4-1035-4edb-85d2-350885d35ea5.png)

I use this, because I send the message options to a queue with the message params to send latter. So the user takes the messageId before the message send, and recive the notification after the message was arrived in device with the same messageId to confirmation.